sdk/trace/batch_span_processor_test.go

```diff
@@ -633,3 +644,235 @@ func TestBatchSpanProcessorConcurrentSafe(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestBatchSpanProcessorMetricsDisabled(t *testing.T) {
+	t.Setenv("OTEL_GO_X_SELF_OBSERVABILITY", "false")
+	tp := basicTracerProvider(t)
+	reader := sdkmetric.NewManualReader()
+	meterProvider := sdkmetric.NewMeterProvider(sdkmetric.WithReader(reader))
+	otel.SetMeterProvider(meterProvider)
+	me := newBlockingExporter()
+	t.Cleanup(func() { assert.NoError(t, me.Shutdown(context.Background())) })
```

Comment:
This code is common for all tests. What do you think of refactoring e.g. into a table-driven test like in https://github.com/open-telemetry/opentelemetry-go/pull/7027/files?